Content/Items/Misc/Weapons.Earthduster.cs

```diff
@@ -0,0 +1,560 @@
+        };
+
+        public override bool CanConsumeAmmo(Item ammo, Player player) => false;
+        public override bool SafeCanUseItem(Player player) => player.ownedProjectileCounts[ModContent.ProjectileType<EarthdusterHoldout>()] <= 0;
+        public override string Texture => AssetDirectory.MiscItem + Name;
+
+        public override void SetStaticDefaults()
+        {
+            Tooltip.SetDefault("Hold <left> to fire a rapid stream of soil\nCan use many different types of soils\n33% chance to not consume ammo");
+        }
+
+        public override void SafeSetDefaults()
+        {
+            Item.DamageType = DamageClass.Ranged;
+            Item.damage = 25;
+            Item.width = 70;
+            Item.height = 40;
+            Item.useAnimation = Item.useTime = 5;
+            Item.shoot = ProjectileID.PurificationPowder;
+            Item.useStyle = ItemUseStyleID.Shoot;
+            Item.rare = ItemRarityID.Orange;
+            Item.shootSpeed = 21f;
+            Item.noUseGraphic = true;
+            Item.noMelee = true;
+            Item.channel = true;
+            Item.knockBack = 3f;
+        }
+
+        public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+        {
+            Projectile proj = Projectile.NewProjectileDirect(source, position, velocity, ModContent.ProjectileType<EarthdusterHoldout>(), damage, knockback, player.whoAmI);
+
+            if (proj.ModProjectile is EarthdusterHoldout earthduster)
+                earthduster.SoilType = type;
+            return false;
+        }
+
+        public override void AddRecipes()
+        {
+            CreateRecipe().
+                AddIngredient(ModContent.ItemType<Soilgun>()).
+                AddIngredient(ItemID.CrimtaneBar, 12).
+                AddIngredient(ItemID.Bone, 50).
+                AddTile(TileID.Anvils).
+                Register();
+
+            CreateRecipe().
+                AddIngredient(ModContent.ItemType<Earthduster>()).
+                AddIngredient(ItemID.CrimtaneBar, 12).
+                AddIngredient(ItemID.Bone, 50).
+                AddTile(TileID.Anvils).
+                Register();
+        }
+    }
+
+    class EarthdusterHoldout : ModProjectile
+    {
+        public const int MAXSHOTS = 100;
+
+        public const int MAXCHARGEDELAY = 30;
+
+        public int SoilType;
+
+        public int shots;
+
+        public int initialShots;
+
+        public float rotTimer;
+
+        public bool charged;
+
+        public bool draw; //only draw two ticks after spawning
+
+        public bool reloading;
+
+        public bool forceReload;
+
+        public Vector2 mouse;
+
+        public ref float ShootDelay => ref Projectile.ai[0];
+
+        public ref float MaxShootDelay => ref Projectile.ai[1];
+
+        public Player owner => Main.player[Projectile.owner];
```

Comment:
Naming rule violation -- Properties should be UpperCamelCase. VS should catch this. Make sure you pay attention to your warnings list and suggestions!